com.unity.render-pipelines.high-definition/Documentation~/Custom-Post-Process.md

```diff
@@ -116,127 +99,90 @@ Now there are the **Setup**, **Render**, and **Cleanup** functions. These are he
-    
     #include "Packages/com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/RTUpscale.hlsl"
     
     struct Attributes
-    
     {
-    
         uint vertexID : SV_VertexID;
-    
         UNITY_VERTEX_INPUT_INSTANCE_ID
-    
     };
     
     struct Varyings
-    
     {
-    
         float4 positionCS : SV_POSITION;
-    
         float2 texcoord   : TEXCOORD0;
-    
         UNITY_VERTEX_OUTPUT_STEREO
     
     };
     
     Varyings Vert(Attributes input)
-    
     {
-    
         Varyings output;
     
         UNITY_SETUP_INSTANCE_ID(input);
-    
         UNITY_INITIALIZE_VERTEX_OUTPUT_STEREO(output);
     
         output.positionCS = GetFullScreenTriangleVertexPosition(input.vertexID);
-    
         output.texcoord = GetFullScreenTriangleTexCoord(input.vertexID);
     
         return output;
-    
     }
     
     // List of properties to control your post process effect
-    
     float _Intensity;
-    
     TEXTURE2D_X(_InputTexture);
     
     float4 CustomPostProcess(Varyings input) : SV_Target
-    
     {
-    
         UNITY_SETUP_STEREO_EYE_INDEX_POST_VERTEX(input);
     
         uint2 positionSS = input.texcoord * _ScreenSize.xy;
-    
         float3 outColor = LOAD_TEXTURE2D_X(_InputTexture, positionSS).xyz;
-    
+
+        // We do the blending manually instead of relying on the hardware blend
+        // It's necessary because the color buffer contains garbage from the previous post process effect.
         return float4(lerp(outColor, Luminance(outColor).xxx, _Intensity), 1);
-    
     }
     
     ENDHLSL
     
     SubShader
-    
     {
-    
         Pass
-    
         {
-    
             Name "GrayScale"
     
             ZWrite Off
-    
             ZTest Always
-    
             Blend Off
-    
             Cull Off
     
             HLSLPROGRAM
-    
                 #pragma fragment CustomPostProcess
-    
                 #pragma vertex Vert
-    
             ENDHLSL
-    
         }
-    
     }
     
     Fallback Off
-
 }
 ```
 
 If none of your Scenes reference the Shader, Unity does not build the Shader and the effect does not work when you run your application outside of the Editor. To resolve this, either add the Shader to a [Resources folder](https://docs.unity3d.com/Manual/LoadingResourcesatRuntime.html), or go to **Edit > Project Settings > Graphics** and add the Shader to the **Always Included Shaders** list.
 
-Note that, if you use the clip() instruction in your Shader, your effect breaks.
+:warning: Note that, when we execute your post process, we use a render target pooling system. It implies that you don't know what the current color buffer contains, which is why we never use any instructions that could show this color buffer. In other words, don't use transparency, blend modes or the the **clip()** instruction in your Shader, otherwise your effect breaks.
```

Comment:
Note that, when HDRP executes your post-process effect, it uses a render target pooling system. It means that you don't know what the current color buffer contains, which is why you should never use any instructions that could show this color buffer. Do not use transparency, blend modes, or the **clip()** instruction in your Shader, otherwise your effect breaks.